www/src/App.js

```diff
@@ -1,21 +1,58 @@
+import 'semantic-ui-css/semantic.min.css';
 import React, { Component } from 'react';
-import Title from './Title';
+import { Link, Route, Switch } from 'react-router-dom';
+import { Menu } from 'semantic-ui-react'
+import Home from './Home';
 
-import logo from './logo.svg';
-import './App.css';
+// helpful routing tutorial
+// https://www.sitepoint.com/react-router-v4-complete-guide/
+// https://css-tricks.com/react-router-4/
+// https://github.com/pillarjs/path-to-regexp#custom-match-parameters
+const MerchantLogin = () => (
+  <div>
+    <h2>Merchant Login</h2>
+  </div>
+)
+
+const MerchantSignup = () => (
+  <div>
+    <h2>Merchant Signup</h2>
+  </div>
+)
+
+const MerchantShop = (props) => {
+  console.log(props);
+
+  return (
+    <div>
+      <h2>Merchant Shop {props.match.params.id}</h2>
+    </div>
+  )
+}
 
 class App extends Component {
   render() {
     return (
-      <div className="App">
-        <header className="App-header">
-          <img src={logo} className="App-logo" alt="logo" />
-          <h1 className="App-title">Welcome to React</h1>
-          <Title>Hello!</Title>
-        </header>
-        <p className="App-intro">
-          To get started, edit <code>src/App.js</code> and save to reload.
-        </p>
+      <div>
+        <Menu inverted>
+          <Menu.Item>Gatsby Manor</Menu.Item>
+          <Menu.Item>
+            <Link to="/">Home</Link>
+          </Menu.Item>
+          <Menu.Item>
+            <Link to="/merchant/login">Login</Link>
+          </Menu.Item>
+          <Menu.Item>
+            <Link to="/merchant/signup">Signup</Link>
+          </Menu.Item>
+        </Menu>
+
+        <Switch>
+          <Route exact path="/" component={Home} />
+          <Route path="/merchant/login" component={MerchantLogin} />
+          <Route path="/merchant/signup" component={MerchantSignup} />
+          <Route path="/merchant/:id" component={MerchantShop} />
+        </Switch>
```

Comment:
Need to take a moment to think how we handle the front end routing. Its a bit confusing on how to compose the routes but so far it seems to be a good idea to keep all top level routes that are gateways to whole different flows under a switch statement so that it only chooses that one path. Then nested routes will take care to show the finer details of the overall scheme. 

Think about the route you want, then build nested routes based on that. 
Based on what I see above maybe the larger abstraction can be to nest login,signup, and :id under merchant since the merchants tag is common to all of them.